cli/src/common.rs

```diff
@@ -129,11 +169,19 @@ where
         },
     };
 
-    let target = ocd_targets::select_target(
-        shared_options.target.as_ref().map(|s| s.as_ref())
-    )?;
+    let target = get_checked_target(shared_options.target.clone());
```

Comment:
Returning an error here allows centralized error handling for all CLI commands in cli/src/main.c, and then the get_checked_target functions is not needed.